homeassistant/components/zha/sensor.py

```diff
@@ -278,6 +299,29 @@ def native_unit_of_measurement(self) -> str:
         """Return Unit of measurement."""
         return self._channel.unit_of_measurement
 
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return device state attrs for battery sensors."""
+        attrs = {}
+        if self._channel.device_type is not None:
+            attrs["device_type"] = self._channel.device_type
+        status = self._channel.status
+        if status is not None:
+            attrs["status"] = str(status)[len(status.__class__.__name__) + 1 :]
+        return attrs
+
+
+@MULTI_MATCH(channel_names=CHANNEL_SMARTENERGY_METERING)
+class SmartEnergySummation(SmartEnergyMetering, id_suffix="summation_delivered"):
+    """Smart Energy Metering summation sensor."""
+
+    SENSOR_ATTR: int | str = "current_summ_delivered"
```

Comment:
Possibly you need 
`_device_class: str | None = DEVICE_CLASS_ENERGY`
maybe also 
`_unit = ENERGY_KILO_WATT_HOUR`
here.  When I connect to my device, the new metering sensor appears (great!), but it is displayed in W (should be Wh or kWh).

Reply:
It is summation value. It cannot be KWh. And unit of measurement is W is set by zigbee channel.

Reply:
ZCL Specs are confusing at times. Summation is:
![image](https://user-images.githubusercontent.com/5826160/134996036-2c797552-7148-4609-8ae6-0bd30dce0dde.png)

and unit of measurement attribute both indicates kWh and kW 
![image](https://user-images.githubusercontent.com/5826160/134996318-48164b7b-be24-450c-b738-0c162d81dc51.png)

I guess my questions then are:
1. is summation supposed to be kW*h or just kW?
2. In case this is a non electric meter (water, gas etc) is it still `DEVICE_CLASS_ENERGY`?

Reply:
If it is a smart energy sensor measuring summation, then I suppose the class will always be energy.

Gas can be measured in kWh but usually m^3 where I live.  It feels like water should be something else.

Reply:
Ok, made the summation sensor a `kWh` UOM if is an electric measurement and ignore device specified formatting, otherwise fallback to the channel UOM and formatting

Reply:
> 
> 
> If it is a smart energy sensor measuring summation, then I suppose the class will always be energy.
> 
> Gas can be measured in kWh but usually m^3 where I live. It feels like water should be something else.

Yes, gas and water consumption is typically measured by volume; e.g., in m^3. Gas consumption can also be measured in, e.g., kWh or J, but the conversion factor would depend on the type of gas. 

I suppose the issue here is that while HA defines different device classes for ENERGY and POWER which are suitable for electrical summation and demand measurements, respectively, HA does not seem to provide classes corresponding to VOLUME and FLOW, which could be used for gas and water. 

There is a GAS device class, but I am not sure if it correspond to volume or flow. Maybe it does not differentiate and differentiation for gas is by uom? Furthermore HA does not seem to provide a generic device class for electric devices or water.